polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -4992,7 +4992,8 @@ mlir::FunctionOpInterface MLIRASTConsumer::GetOrCreateMLIRFunction(
       attrs.set("llvm.linkage",
                 mlir::LLVM::LinkageAttr::get(builder.getContext(), lnk));
       function->setAttrs(attrs.getDictionary(builder.getContext()));
-      if (ShouldEmit) {
+      // Do not add to functionsToEmit if it has already been initialized.
```

Comment:
We probably don't need this change after moving `done.emplace(...).second` check before invoking `GetOrCreateMLIRFunction`.

Reply:
Indeed, the previous PR allows that.